### DISCUSSION TITLE
How to do a conditional return in withObservables with a non-reactive value?

### DISCUSSION THREAD

**Discussion by Albert-Gao** [Nozbe/WatermelonDB#1427] [Q&A] [answered]
So, I have a user model, and have an `assignedTasks()` query on it, now I want to render a component with the count.

Here is how I do it. Since I need to handle the case which the `currentUser` is null. I was doing the step A with `fetchAndObserve()`, but it throws error when record not found and seems uncatchable, so I switched to a normal `Q.where()`, it indeed took me to the step B, now I have another problem, since the currentUser could be not there, I need to return a fallback count, otherwise, invoking `currentUser[0].assignedTasks()` would cause a problem like `can not invoke assignedTasks on undefined`.

How to solve this problem? Or I would say what is the `watermelon` way of solving it? Thanks :)

```typescript
const enhance = compose<any, any>(
  // step A
  withObservables([], ({ database }: { database: Database }) => ({
    currentUser: database.collections
      .get(Tables.Users)
      .query(Q.where("id", CurrentUser.userId()))
      .observe(),
  })),
  
  // step B
  withObservables(
    ["currentUser"],
    ({ currentUser }: { currentUser: User[] }) => ({
      assignedToMeCount: isEmpty(currentUser)
        ? 0
        : currentUser[0].assignedTasks().observeCount(),
    })
  )
);
```

**Comment by radex** [comment, marked as answer]
Simply return `foo ? Rx.of(0) : query.observeCount()`

https://rxjs-dev.firebaseapp.com/api/index/function/of

**Reply by Albert-Gao** [reply]
Thanks so much, works beautifully now :)

**Reply by Albert-Gao** [reply]
is there a way to make this query return a type like `User | null` instead of the `User[]`?

```typescript
database.collections
      .get(Tables.Users)
      .query(Q.where("id", CurrentUser.userId()))
      .observe()
```

not a rxjs guy, thus asking :) thanks in advance

**Reply by radex** [reply]
You can unwrap the array inside the component, you don't need to do it via RxJS.

But you can if you want to. Here's a snippet from @Nozbe you could adapt to your needs

<img width="491" alt="image" src="https://user-images.githubusercontent.com/183747/199484019-8fa90d86-f0cc-413a-ac72-b30fb648dc03.png">
